cpp/include/cudf/detail/copy_if.cuh

```diff
@@ -292,10 +295,13 @@ struct scatter_gather_functor {
                     indices.begin(),
                     filter);
 
+    auto gather_map = column_view(
+      cudf::data_type{cudf::type_to_id<cudf::size_type>()}, output_size, indices.data());
```

Comment:
If we're going to go this route, we should have a `column_view` ctor from a `device_uvector` or `device_span` instead of the error prone manual construction everywhere.

Reply:
I was thinking more of a factory or utility function. What makes me nervous is that `device_uvector` and `device_span` size types are `size_t` and so technically those could be bigger than what `column_view` allows. And this makes me want to add a `CUDF_EXPECTS` check which I'd rather not do in a ctor.

Reply:
I don't have too many qualms about throwing in a ctor. No different than `strings_column_view` ctor throwing if the incoming `column_view` is a string type.

Reply:
Another option that I'd prefer is to make a `detail::gather()` call accept a `device_span` as `gather_map` argument. This requires the least amount of changes.

Reply:
I like that idea. We can just put the `CUDF_EXPECTS` in that overload and then call the `column_view` overload.